api/routes.go

```diff
@@ -3,19 +3,23 @@ package api
 import (
 	"context"
 	"net/http"
-	"time"
 
 	"github.com/julienschmidt/httprouter"
 )
 
 // buildHTTPRoutes registers all HTTP routes and their handlers.
 func (api *API) buildHTTPRoutes() {
+	api.staticRouter.POST("/login", api.loginHandler)
```

Comment:
This endpoint also requires validate right? It should validate whether the authorisation header is provided and valid before logging user in.

```suggestion
	api.staticRouter.POST("/login", api.validate(api.loginHandler))
```

Reply:
The endpoint does validate internally. It basically repeats the steps `validate` does but keeps the string representation of the token in order to put it in the cookie. I've removed the `validate` here in order to avoid doing it twice.

Reply:
you shouldn't do it in the endpoint again if you validate it here I guess but maybe I didn't read thorough enough, I'll leave it to a second reviewer